src/LibraryUtilities.ts

```diff
@@ -103,17 +105,12 @@ export function constructNestedLibraryItems(
     for (let i = startIndex; i < fullNameParts.length; i++) {
         let libraryItem = new ItemData(fullNameParts[i]);
         libraryItem.itemType = "none";
-
-        // If 'i' is now '2' (i.e. it points to 'C'), then we will construct 
-        // the iconName as 'A.B.C'. And since the second parameter of 'slice' 
-        // is exclusive, we add '1' to it otherwise 'C' won't be included.
-        //  
-        libraryItem.iconName = fullNameParts.slice(0, i + 1).join(".");
+        libraryItem.iconUrl = iconUrl; 
```

Comment:
We can keep the original logic and use it just in case `iconUrl` is not provided:

```js
if (iconUrl) {
    libraryItem.iconUrl = iconUrl;
} else {
    // If 'i' is now '2' (i.e. it points to 'C'), then we will construct 
    // ...
    libraryItem.iconName = fullNameParts.slice(0, i + 1).join(".");
}
```